src/storage/index.rs

```diff
@@ -0,0 +1,226 @@
+use std::fs::File;
+use std::io;
+use std::io::prelude::*;
+use std::io::Seek;
+use std::io::SeekFrom;
+use std::mem;
+
+// meta data of raw table
+struct TableMeta {
+    table_name: String, // name of raw table
+    key_type: String,   // type of primary key in raw table
+    key_offet: u32,     // byte position of first primary key in raw table
+    key_bytes: u32,     // bytes of primary key in raw table
+    row_bytes: u32,     // bytes of each row in raw table
+}
+
+// row and key value pair in which key type is int
+struct IndexDataStructureInt {
+    row: u32,
+    key_value: u32,
+}
+// row and key value pair in which key type is string
+struct IndexDataStructureString {
+    row: u32,
+    key_value: Vec<u8>,
+}
+
+// build index table with raw table in whicj key type is int
+fn build_int_index_table(table_meta: &TableMeta, index_arr: &mut Vec<IndexDataStructureInt>) {
```

Comment:
let's make `Vec<IndexDataStructureInt>)` be a type `Index(Vec<IndexDataStructureInt>)>`, so other can use it easily.